program-runtime/src/invoke_context.rs

```diff
@@ -499,7 +505,8 @@ impl<'a> InvokeContext<'a> {
             &program_indices,
             &account_indices,
             &caller_write_privileges,
-        )?;
+        )
+        .result?;
```

Comment:
not sure if the compute budget is updated somewhere earlier/later here, and whether this is subject to the same bug. Will look further.